lib/zendesk_api/client.rb

```diff
@@ -40,6 +40,8 @@ def method_missing(method, *args, &block)
       method = method.to_s
       options = args.last.is_a?(Hash) ? args.pop : {}
 
+      return ZendeskAPI::Collection.new(self, method_as_class(method), options) unless config.use_resource_cache
```

Comment:
I think this is missing the nil check on `method_as_class` from line 50:

```
raise "Resource for #{method} does not exist" unless @resource_cache[method][:class]
```

unless `ZendeskAPI::Collection#initialize` is also doing a nill-check on the `method_as_class` argument?

Reply:
Good catch! Will fix